flopy/modflowusg/mfusgcln.py

```diff
@@ -0,0 +1,1106 @@
+            + "but received type: {type(model)}."
+        )
+        assert isinstance(model, ModflowUsg), msg
+
+        valid_args_defaults = {
+            "ncln": None,  # number of CLNs
+            "iclnnds": None,  # number of nodes
+            "nndcln": None,  # number of nodes in each CLN segments
+            "clncon": None,  # node IDs in each CLN segments
+            "nja_cln": None,  # total number of node-node connections (NJAG)
+            "iac_cln": None,  # number of connections for each node (sum(IAC)": NJAG
+            "ja_cln": None,  # node connections
+            "node_prop": None,  # node properties
+            "nclngwc": None,  # number of CLN-GW connections
+            "cln_gwc": None,  # CLN-GW connections
+            "nconduityp": 1,  # number of circular conduit types
+            "cln_circ": None,  # circular conduit properties
+            "ibound": 1,  # boundary condition types
+            "strt": 1.0,  # initial head in CLN cells
+            "transient": False,  # OPTIONS: transient IBOUND for each stress period
+            "printiaja": False,  # OPTIONS: print IA_CLN and JA_CLN to listing file
+            "nrectyp": 0,  # OPTIONS2: number of rectangular fracture types
+            "cln_rect": None,  # rectangular fracture properties
+            "bhe": False,  # OPTIONS2: borehole heat exchanger (bhe)
+            "grav": None,  # OPTIONS2: gravitational acceleration constant
+            "visk": None,  # OPTIONS2: kinematic viscosity of water
+            "extension": [
+                "cln",
+                "clncb",
+                "clnhd",
+                "clndd",
+                "clnib",
+                "clncn",
+                "clnmb",
+            ],
+            "unitnumber": None,
+            "filenames": None,
+        }
+
+        for arg, default_value in valid_args_defaults.items():
+            setattr(self, arg, kwargs.pop(arg, default_value))
+
+        # set default unit number of one is not specified
+        if self.unitnumber is None:
+            self.unitnumber = ModflowUsgCln._defaultunit()
+        elif isinstance(self.unitnumber, list):
+            if len(self.unitnumber) < 7:
+                for idx in range(len(self.unitnumber), 7):
+                    self.unitnumber.append(0)
+
+        # set filenames
+        if self.filenames is None:
+            self.filenames = [None, None, None, None, None, None, None]
+        elif isinstance(self.filenames, str):
+            self.filenames = [
+                self.filenames,
+                None,
+                None,
+                None,
+                None,
+                None,
+                None,
+            ]
+        elif isinstance(self.filenames, list):
+            if len(self.filenames) < 7:
+                for idx in range(len(self.filenames), 7):
+                    self.filenames.append(None)
+
+        # Fill namefile items
+        name = [ModflowUsgCln._ftype()]
```

Comment:
Within class methods like `__init__(self, ...`, this can simply be `self._ftype()`. Classmethods like `load(cls, ...) it can be `cls._ftype()`. There ~17 "._ftype()" matches here  that could potentially be simplified.